### PR TITLE
infoschema, planner, executor: add virtual system table `information_schema.inspection_rules`

### PR DESCRIPTION
**Pull request by dasinlsb**
UCP #15013 

### What problem does this PR solve? 

add virtual system table `information_schema.inspection_rules`

### What is changed and how it works?

+ add virtual system table `information_schema.inspection_rules` schema: NAME, TYPE('inspection'/'summary').

+ implement `inspectionRuleTableExtractor` to filter column `TYPE`.

+ implement `inspectionRuleRetriever` to return rule names stored in memory.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

`inspectionRules` is exported in `executor/executor_pkg_test.go` for test

### CODE REVIEW

File: executor/inspection_common.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (djshow832) [inline executor/inspection_common.go]:
```suggestion
// Copyright 2020 PingCAP, Inc.
```

File: executor/inspection_common_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (djshow832) [inline executor/inspection_common_test.go]: Ditto.

File: infoschema/tables.go [hunk 3]
```diff
@@ -929,6 +932,11 @@ var tableInspectionSummaryCols = []columnInfo{
 	{name: "MAX_VALUE", tp: mysql.TypeDouble, size: 22, decimal: 6},
 }
 
+var tableInspectionRulesCols = []columnInfo{
+	{name: "NAME", tp: mysql.TypeVarchar, size: 64},
+	{name: "TYPE", tp: mysql.TypeVarchar, size: 64},
```
Comment (lonng) [inline infoschema/tables.go]: It's better to add a column `COMMENT`, we may add some explanation for the rule. （You can open a new pr to add an explanation to each rule）

Comment (dasinlsb) [reply infoschema/tables.go]: now I add a column `COMMENT` leaving empty ( I should read docs later to understand those rules..

Comment (lonng) [reply infoschema/tables.go]: Thanks

**Pull request merged**